src/protocol.cpp

```diff
@@ -85,8 +85,13 @@ CMessageHeader::CMessageHeader(const MessageStartChars& pchMessageStartIn)
 CMessageHeader::CMessageHeader(const MessageStartChars& pchMessageStartIn, const char* pszCommand, unsigned int nMessageSizeIn)
 {
     memcpy(pchMessageStart, pchMessageStartIn, MESSAGE_START_SIZE);
-    memset(pchCommand, 0, sizeof(pchCommand));
-    strncpy(pchCommand, pszCommand, COMMAND_SIZE);
+
+    // Copy the command name, zero-padding to COMMAND_SIZE bytes
+    size_t i = 0;
+    for (; i < COMMAND_SIZE && pszCommand[i] != 0; ++i) pchCommand[i] = pszCommand[i];
+    assert(pszCommand[i] == 0); // Assert that the command name passed in is not longer than COMMAND_SIZE
+    for (; i < COMMAND_SIZE; ++i) pchCommand[i] = 0;
```

Comment:
In commit "net: Fail instead of truncate command name in CMessageHeader" (b837b334db5dd6232725fd2350928ff4fbd3feee)

I don't want to second guess your code style preferences, but I believe you could just add  parentheses around strncpy to silence the warning. IMO, following would be the simplest way of filling the buffer:

```c++
(strncpy(pchCommand, pszCommand, COMMAND_SIZE));
assert(pchCommand[COMMAND_SIZE-1] == '\0');
```

Definitely good to drop the memset in either case. That is completely pointless and doesn't do anything.

Reply:
@ryanofsky The memset used to be necessary to zero-pad the copy, no?

Reply:
I mean—this is pretty trivial code to review, only a few bytes are copied, no high performance platform specific implementation is needed.
I don't think we need to rely on any C string functions here. But if you prefer that, sure…